### PR TITLE
Add partner & stress network config

### PR DESCRIPTION
**Pull request by Daniel-VDM**
* Add partner config to main.go
* Add partner config internals
* Refactor and add partner logic to genesis node

Partner config is needed for PSTN

### CODE REVIEW

File: internal/configs/sharding/partner.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	default: // genesis
+		return partnerV0
+	}
+}
+
+func (partnerSchedule) BlocksPerEpoch() uint64 {
+	return partnerBlocksPerEpoch
+}
+
+func (ts partnerSchedule) CalcEpochNumber(blockNum uint64) *big.Int {
```
Comment (LeoHChen) [inline internal/configs/sharding/partner.go]: better us ps

File: internal/configs/sharding/partner.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+func (ts partnerSchedule) GetShardingStructure(numShard, shardID int) []map[string]interface{} {
+	return genShardingStructure(numShard, shardID, PartnerHTTPPattern, PartnerWSPattern)
+}
+
+var partnerReshardingEpoch = []*big.Int{
+	big.NewInt(0),
+	params.TestnetChainConfig.StakingEpoch,
+}
+
+var partnerV0 = MustNewInstance(2, 10, 10, genesis.TNHarmonyAccounts, genesis.TNFoundationalAccounts, partnerReshardingEpoch, PartnerSchedule.BlocksPerEpoch())
```
Comment (LeoHChen) [inline internal/configs/sharding/partner.go]: we can go a bit higher, considering we use multi-key configuration.  maybe 2*15, 3 keys per node and we have 5 nodes per shard.

File: internal/configs/sharding/partner.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+var partnerV1 = MustNewInstance(2, 20, 10, genesis.TNHarmonyAccounts, genesis.TNFoundationalAccounts, partnerReshardingEpoch, PartnerSchedule.BlocksPerEpoch())
```
Comment (LeoHChen) [inline internal/configs/sharding/partner.go]: 2, 30, 15.

File: node/node_genesis.go [hunk 4]
```diff
@@ -86,9 +88,14 @@ func (node *Node) SetupGenesisBlock(db ethdb.Database, shardID uint32, myShardSt
 		}
 	case nodeconfig.Pangaea:
 		chainConfig = *params.PangaeaChainConfig
-		fallthrough // the rest is the same as testnet
+	case nodeconfig.Partner:
```
Comment (LeoHChen) [inline node/node_genesis.go]: why don't you add nodeconfig.Testnet here as well?

File: internal/configs/sharding/partner.go (hunk 1 above)
Comment (Daniel-VDM) [reply internal/configs/sharding/partner.go]: fixed in latest

File: node/node_genesis.go (hunk 4 above)
Comment (Daniel-VDM) [reply node/node_genesis.go]: Testnet is the default config for all of the other chain configs.

File: internal/configs/sharding/partner.go (hunk 2 above)
Comment (Daniel-VDM) [reply internal/configs/sharding/partner.go]: updated

File: internal/configs/sharding/partner.go (hunk 3 above)
Comment (Daniel-VDM) [reply internal/configs/sharding/partner.go]: updated

**Comment by Daniel-VDM** [comment]
Also added config for stressnet

File: internal/configs/sharding/stress.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+type stressnetSchedule struct{}
+
+const (
+	// 10 minutes per epoch (at 8s/block)
+	stressnetBlocksPerEpoch = 75
+
+	stressnetVdfDifficulty = 10000 // This takes about 20s to finish the vdf
+
+	// StressNetHTTPPattern is the http pattern for stressnet.
+	StressNetHTTPPattern = "https://api.s%d.os.hmny.io"
```
Comment (LeoHChen) [inline internal/configs/sharding/stress.go]: os => stn

File: internal/configs/sharding/stress.go [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+func (ss stressnetSchedule) GetShardingStructure(numShard, shardID int) []map[string]interface{} {
+	return genShardingStructure(numShard, shardID, StressNetHTTPPattern, StressNetWSPattern)
+}
+
+var stressnetReshardingEpoch = []*big.Int{
+	big.NewInt(0),
+	params.StressnetChainConfig.StakingEpoch,
+}
+
+var stressnetV0 = MustNewInstance(4, 27, 27, genesis.TNHarmonyAccounts, genesis.TNFoundationalAccounts, stressnetReshardingEpoch, StressNetSchedule.BlocksPerEpoch())
```
Comment (LeoHChen) [inline internal/configs/sharding/stress.go]: we can have a larger stress test net. it's a short live network anyway. how about 4, 150, 120, with 3 keys per node

File: internal/configs/sharding/stress.go (hunk 5 above)
Comment (Daniel-VDM) [reply internal/configs/sharding/stress.go]: changed

File: internal/configs/sharding/stress.go (hunk 6 above)
Comment (Daniel-VDM) [reply internal/configs/sharding/stress.go]: sure changed to suggesstion

**Pull request merged**